src/os/user/read_colon_file.go

```diff
@@ -0,0 +1,86 @@
+//go:build !windows
+// +build !windows
+
+package user
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+)
+
+// lineFunc returns a value, an error, or (nil, nil) to skip the row.
+type lineFunc func(line []byte) (v interface{}, err error)
+
+// readColonFile parses r as an /etc/group or /etc/passwd style file, running
+// fn for each row. readColonFile returns a value, an error, or (nil, nil) if
+// the end of the file is reached without a match.
+//
+// readCols is the minimum number of colon-separated fields that will be passed
+// to fn; in a long line additional fields may be silently discarded.
+//
+// readColonFile can also be used to read /adm/users on plan9.
+func readColonFile(r io.Reader, fn lineFunc, readCols int) (v interface{}, err error) {
```

Comment:
does it have to move to a separate file? Moving it to a separate file makes it hard to see what's changed between the old & new versions.

Reply:
Yes, we need it on plan9, because it does not make sense to create another function which does exactly the same thing. Since originally, this function was in `lookup_unix.go`, which also contains lookup implementation for non-cgo unix version, we need a separate file for this.

Reply:
Move this to different commit